### DISCUSSION TITLE
How to add button (Menu) in user input field?

### DISCUSSION THREAD

**Discussion by code-by** [telegraf/telegraf#1747] [Q&A] [answered]
I'd like to add commands Menu in telegram like here
https://stackoverflow.com/questions/69158820/how-to-create-a-three-dashes-menu-on-a-telegram-bot
or
[here](https://user-images.githubusercontent.com/84489925/169064383-c7554fdc-d08e-4b6c-9d81-29370886d900.jpeg)
using telegraf instead set commands list manually in BotFather. It is possible?

**Comment by MKRhere** [comment, marked as answer]
This only works for private chats:

```TS
// from a handler
ctx.setChatMenuButton(menuButton);

// outside a handler
bot.telegram.setChatMenuButton(chatId, menuButton);
```

See the docs: https://core.telegram.org/bots/api#setchatmenubutton

**Reply by sunnyplayzap** [reply]
What goes in menuButton? can you please give proper example? I am trying to create a game button with custom parameters

**Reply by DoctorLeiws** [reply]
I have same problem, how to input custom number into a button

**Reply by sunnyplayzap** [reply]
So after doing hell of a research i figured it out,
Here's the demo code for custom MenuButton : 
 ```
 await ctx.setChatMenuButton(JSON.stringify({
    type: 'web_app',
    text: '🕹️ Open app',
    web_app: { url: "your_game_url" }
  }))
```
<img width="387" alt="Screenshot 2024-01-26 at 11 29 30 AM" src="https://github.com/telegraf/telegraf/assets/138470562/56a2d02b-66f3-4dbe-8f41-4be016c07d69">

**Reply by MKRhere** [reply]
You do not need to do `JSON.stringify`. Just the object will work.

**Reply by 7heMech** [reply]
Can you make it execute a function on the bot instead of a webapp?

**Comment by addreeh** [comment]
I have set that:
![image](https://github.com/telegraf/telegraf/assets/74270582/a952ccd3-c096-4cb8-b696-64fe4868cfb6)


The code dont fail buy I dont see the button:
```
const menuButton: MenuButton = {
  type: 'web_app',
  text: 'Open app',
  web_app: {
    url: 'https://example.com/your_game_url'
  }
}

bot.telegram.setChatMenuButton({ chatId: -1001772146274, menuButton })
  .then(() => console.log('Menu button set successfully'))
  .catch((error) => console.error('Error setting chat menu button:', error))

bot.on('message', unknown())
```

**Reply by sunnyplayzap** [reply]
setChatMenuButton doesn't work in group chats, if there's a - (negative) in chatId that means you're in a group chat

**Reply by agusArtana** [reply]
I'm using a telegram bot api library, and have set a menu button via botfather and i want to configure it via coding but i don't know how to do it as I don't really understand the documentation. Please give me an advise, i use express.js

**Comment by felixcary** [comment]
Hi, anyone can help why the button not showing in my case? @MKRhere maybe?
<img width="701" alt="image" src="https://github.com/user-attachments/assets/cd888c14-bccc-47f6-beb4-202a412c0d9e">
<img width="387" alt="image" src="https://github.com/user-attachments/assets/a1c2fcd3-d0a6-4935-9744-59534f3a94d1">

**Comment by ngxvu** [comment]
![image](https://github.com/user-attachments/assets/d052d3c3-4439-49a6-a3c0-6c67e985eb8a)

using /setmenubutton
